src/main/java/com/github/krzysiekjodlowski/ox/ui/UI.java

```diff
@@ -0,0 +1,27 @@
+package com.github.krzysiekjodlowski.ox.ui;
+
+import com.github.krzysiekjodlowski.ox.NumberRange;
+
+/**
+ * Interacting directly with user,
+ * used for collecting input end
+ * outputting messages to user.
+ *
+ * @author Krzysztof Jodlowski
+ */
+public interface UI<T1, T2 extends Number> {
+  /**
+   * Outputs message to user.
+   *
+   * @param message any appropriate representation
+   */
+  void say(T1 message);
+
+  /**
+   * Returns Number type from user input.
+   *
+   * @param range custom Number type
+   * @return custom Number type
+   */
+  T2 getNumberFromUser(NumberRange<T2> range);
+}
```

Comment:
In my opinion too few methods. You can only show a message and get number from input. No option for showing error for Console UI to error stream for example (it could be error method and in console ui implementation could the same as say() but with printing to System.err not System.out).

Reply:
Thats a neat idea! I'll add error showing to UI:)